vars/neoDeploy.groovy

```diff
@@ -19,33 +19,103 @@ def call(parameters = [:]) {
             error "Archive cannot be found with parameter archivePath: '${archivePath}'."
         }
 
+        def deployMode = utils.getMandatoryParameter(parameters, 'deployMode', 'MTA')
+
+        def propertiesFile
+        def warAction
+        if (deployMode == 'WAR_PROPERTIESFILE') {
+            propertiesFile = new File(utils.getMandatoryParameter(parameters, 'propertiesFile', null))
+            if (!propertiesFile.isAbsolute()) {
+                propertiesFile = new File(pwd(), propertiesFile.getPath())
+            }
+            if (!propertiesFile.exists()){
+                error "Properties file cannot be found with parameter propertiesFile: '${propertiesFile}'."
+            }
+            warAction = utils.getMandatoryParameter(parameters, 'warAction', 'deploy')
+            if (warAction != 'warAction' || warAction != 'deploy') {
+                warAction = 'deploy'
+            }
+        }
+
+        def applicationName
+        def runtime
+        def runtimeVersion
+        def vmSize
+        if (deployMode == 'WAR_PARAMS') {
+            applicationName = utils.getMandatoryParameter(parameters, 'applicationName', null)
+            runtime = utils.getMandatoryParameter(parameters, 'runtime', null)
+            runtimeVersion = utils.getMandatoryParameter(parameters, 'runtimeVersion', null)
+            vmSize = utils.getMandatoryParameter(parameters, 'vmSize', 'lite')
+            if (vmSize != 'lite' || vmSize !='pro' || vmSize != 'prem' || vmSize != 'prem-plus') {
+                vmSize = 'lite'
+            }
+            warAction = utils.getMandatoryParameter(parameters, 'warAction', 'deploy')
+        }
+
         def defaultDeployHost = script.commonPipelineEnvironment.getConfigProperty('DEPLOY_HOST')
         def defaultDeployAccount = script.commonPipelineEnvironment.getConfigProperty('CI_DEPLOY_ACCOUNT')
         def defaultCredentialsId = script.commonPipelineEnvironment.getConfigProperty('neoCredentialsId')
         if (defaultCredentialsId == null) {
             defaultCredentialsId = 'CI_CREDENTIALS_ID'
         }
 
-        def deployHost = utils.getMandatoryParameter(parameters, 'deployHost', defaultDeployHost)
-        def deployAccount = utils.getMandatoryParameter(parameters, 'deployAccount', defaultDeployAccount)
+        def deployHost
+        def deployAccount
+
+        if (!deployMode.equals('MTA') && !deployMode.equals('WAR_PARAMS') && !deployMode.equals('WAR_PROPERTIESFILE')) {
```

Comment:
[we should discuss] In the light of "fail fast": wouldn't it be better to fail the build in case of an inappropriate value? People might be unaware of the missing deployment and recognize a missing deployment at some later point in time.